### ISSUE TITLE
Codebase quality

### ISSUE THREAD

**Issue by d2dyno1** [taj-ny/kwin-effects-forceblur#151]
Hi! First off, good job on the project!
I was scouring through the code and came to the conclusion that the primary area for improvement would be to use more abstractions to better support multiple platforms (e.g., other Linux distros, Windows). Right now, there’s quite a bit of platform-specific code that could be abstracted into interfaces or services to make it easier to extend compatibility and maintainability.

Simply put, there aren't enough abstractions. This would help future-proof the project and make scaling to new platforms much smoother.
